internal/api/role_assignment.go

```diff
@@ -0,0 +1,430 @@
+// Copyright (c) 2025 - Stacklet, Inc.
+
+package api
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/hasura/go-graphql-client"
+)
+
+// RoleAssignment is the data returned by reading role assignment data.
+type RoleAssignment struct {
+	ID        string
+	Role      Role
+	Principal rolePrincipal `graphql:"principal"`
+	Target    roleTarget    `graphql:"target"`
```

Comment:
I think we should be able to avoid the principal/target identifiers packing/unpacking by just getting the `roleAssignmentPrincipal`/`roleAssignmentTarget` here as well, so basically something like

```go
type rolePrincipalPrincipal struct {
  RoleAssignmentPrincipal string `graphql:"roleAssignmentPrincipal"`
}

type rolePrincipal struct {
  User     *rolePrincipalPrincipal `graphql:"... on User"`
  SSOGroup *rolePrincipalPrincipal `graphql:"... on SSOGroup"`
}

type roleTarget struct {
  RoleAssignmentTarget string `graphql:"roleAssignmentTarget"`
  ...
}
```

so that we only need to get the not-empty opaque ID across types of principals